alma/lib/Forms/ShareOfCheckoutAdminFormBuilder.php

```diff
@@ -37,14 +37,21 @@ class ShareOfCheckoutAdminFormBuilder extends AbstractAlmaAdminFormBuilder
 {
     const ALMA_ACTIVATE_SHARE_OF_CHECKOUT = 'ALMA_ACTIVATE_SHARE_OF_CHECKOUT';
     const ALMA_SHARE_OF_CHECKOUT_DATE = 'ALMA_SHARE_OF_CHECKOUT_DATE';
+    const ALMA_SHARE_OF_CHECKOUT_CONSENT_NO_ANSWER = 'no_answer';
+    const ALMA_SHARE_OF_CHECKOUT_CONSENT_YES = '1';
+    const ALMA_SHARE_OF_CHECKOUT_CONSENT_NO = '0';
 
     protected function configForm()
     {
         $tpl = $this->context->smarty->createTemplate(
             "{$this->module->local_path}views/templates/hook/_partials/shareOfCheckout.tpl"
         );
 
-        if ((Settings::isShareOfCheckoutSetting() === false && Settings::getActiveMode() === ALMA_MODE_LIVE) || Settings::getActiveMode() !== ALMA_MODE_LIVE) {
+        if (
+            (Settings::isShareOfCheckoutNoAnswered() && Settings::getActiveMode() === ALMA_MODE_LIVE)
+            || (!Settings::isShareOfCheckoutSetting() && Settings::getActiveMode() === ALMA_MODE_LIVE)
+            || Settings::getActiveMode() !== ALMA_MODE_LIVE
```

Comment:
Could we have a `Settings` method for this long test?

something like `Settings::shouldShareOfCheckoutBeDisplayed` or `shouldShareOfCheckoutRun`